src/lib/containers/EvaluationIdRenderer.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react'
+import { useInView } from 'react-intersection-observer'
+import { getEvaluation } from '../utils/SynapseClient'
+import { useSynapseContext } from '../utils/SynapseContext'
+
+export type EvaluationIdRendererProps = {
+  evaluationId: string
+}
+
+const EvaluationIdRenderer: React.FC<EvaluationIdRendererProps> = props => {
+  const { accessToken } = useSynapseContext()
+  const { evaluationId } = props
+  const [evaluationName, setEvaluationName] = useState<string>()
+  const { ref, inView } = useInView()
+  let mounted: boolean = true
+
+  useEffect(() => {
+    if (mounted && inView) {
+      getEvaluationQueue()
+    }
+    return () => {
+      mounted = false
+    }
```

Comment:
I keep seeing this pattern, but does it actually do anything? The return function only runs when the component unmounts, but is it even possible for the effect to run after unmount?

If we need to cancel the request if an unmount happens before the request resolves, then I think we would need to send a signal via AbortController